src/ui/components/createUser/CreatePasswordlessUser.tsx

```diff
@@ -0,0 +1,247 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useContext, useEffect, useState } from "react";
+import { PasswordlessContactMethod } from "../../../api/tenants/login-methods";
+import useCreateUserService, { CreatePasswordlessUserPayload } from "../../../api/user/create";
+import { getApiUrl, getImageUrl } from "../../../utils";
+import { PopupContentContext } from "../../contexts/PopupContentContext";
+import Button from "../button";
+import { Dialog, DialogContent, DialogFooter } from "../dialog";
+import InputField from "../inputField/InputField";
+import { PhoneNumberInput } from "../phoneNumber/PhoneNumberInput";
+import { CreateUserDialogStepType } from "./CreateUserDialog";
+
+type CreatePasswordlessUserProps = {
+	tenantId: string;
+	authMethod: PasswordlessContactMethod;
+	onCloseDialog: () => void;
+	setCurrentStep: (step: CreateUserDialogStepType) => void;
+	loadCount: () => void;
+};
+
+function isNumber(value: string): boolean {
+	const trimmedString = value.replaceAll(/\s/g, "").trim();
+
+	// added this check since parsing a empty string to a number returns 0.
+	if (trimmedString.length < 1) {
+		return false;
+	}
+
+	return isNaN(Number(trimmedString)) === false;
+}
+
+export default function CreatePasswordlessUser({
+	tenantId,
+	authMethod,
+	onCloseDialog,
+	setCurrentStep,
+	loadCount,
+}: CreatePasswordlessUserProps) {
+	const [email, setEmail] = useState("");
+	const [phoneNumber, setPhoneNumber] = useState("");
+	const [emailOrPhone, setEmailOrPhone] = useState("");
+
+	const [generalErrorMessage, setGeneralErrorMessage] = useState<string | undefined>(undefined);
+
+	const [isCreatingUser, setIsCreatingUser] = useState(false);
+	const [isPhoneNumber, setIsPhoneNumber] = useState(false);
+
+	const { createPasswordlessUser } = useCreateUserService();
+	const { showToast } = useContext(PopupContentContext);
+
```

Comment:
400 or 500 errors when we create a user must be shown as something went wrong within the popup, or as a side notification. Not as a full page error boundary.

Reply:
done